loki_hook_options.go

```diff
@@ -57,6 +63,14 @@ type LokiHookOptions interface {
 	// WithFormatter allows for a custom logrus.Formatter to be used when writing log entries
 	// By default, the logrus.TextFormatter will be used
 	WithFormatter(logrus.Formatter) LokiHookOptions
+
+	// WithBasicAuth allows to set a username and password to use when writing to the remote Loki host.
+	WithBasicAuth(username, password string) LokiHookOptions
+}
+
+type basicAuthCredentials struct {
+	username string
+	password string
```

Comment:
nit: Since we can return this via the `BasicAuth()` method, let's make this public